apps/performance-tests/src/unified-selection/HiliteSet.test.ts

```diff
@@ -0,0 +1,117 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { expect } from "chai";
+import { SnapshotDb } from "@itwin/core-backend";
+import { SchemaContext, SchemaJsonLocater } from "@itwin/ecschema-metadata";
+import { createECSchemaProvider, createECSqlQueryExecutor } from "@itwin/presentation-core-interop";
+import { createHiliteSetProvider, Selectable, Selectables } from "@itwin/unified-selection";
+import { Datasets, IModelName } from "../util/Datasets";
+import { run, RunOptions } from "../util/TestUtilities";
+
+describe("hilite", () => {
+  const additionalElementCount = 5;
+  const additionalFunctionalElement2dCount = 1;
+
+  function createSelection({ fullClassName, start, count }: { fullClassName: string; start: number; count: number }) {
+    const selectables: Selectable[] = [];
+    for (let i = start; i < start + count + additionalElementCount; i++) {
+      selectables.push({ className: fullClassName, id: `0x${i.toString(16)}` });
+    }
+    return Selectables.create(selectables);
+  }
+
+  runHiliteTest({
+    testName: "50k elements",
+    iModelName: "50k nested elements",
+    selection: createSelection({ fullClassName: "Generic:PhysicalObject", start: 14, count: 50_000 }),
+    expectedCounts: { elements: 50000 },
+  });
+
+  runHiliteTest({
+    testName: "50k group elements",
+    iModelName: "50k group member elements",
+    selection: createSelection({ fullClassName: "Generic:Group", start: 15, count: 50_000 }),
+    expectedCounts: { elements: 50000 },
+  });
+
+  runHiliteTest({
+    testName: "1k subjects",
+    iModelName: "1k nested subjects",
+    selection: createSelection({ fullClassName: "BisCore:Subject", start: 11, count: 2_000 }),
```

Comment:
What does `count: 2_000` mean if we only have 1k subjects?

Reply:
Since additional elements were inserted between subjects, their ids were not sequential. Instead of doing this, updated `createSelection` function to create keys based on passed `step` value.

Reply:
Please see [my other comment](https://github.com/iTwin/presentation/pull/557#discussion_r1590905205) which, I think, applies here.